core/src/main/java/org/apache/brooklyn/core/mgmt/internal/BasicExternalConfigSupplierRegistry.java

```diff
@@ -103,7 +103,7 @@ private void updateFromBrooklynProperties(ManagementContext mgmt) {
             try {
                 Class<? extends ExternalConfigSupplier> providerClass = (Class<? extends ExternalConfigSupplier>) classloader.loadClass(providerClassname);
                 Constructor<? extends ExternalConfigSupplier> constructor = providerClass.getConstructor(ManagementContext.class, String.class, Map.class);
-                ExternalConfigSupplier configSupplier = constructor.newInstance(this, name, config);
+                ExternalConfigSupplier configSupplier = constructor.newInstance(mgmt, name, config);
```

Comment:
The constructor in `AbstractExternalConfigSupplier` only has `ManagementContext` and `String` arguments. Is it safe to assume that implementors will always provide an implementation with the third `Map` argument?